dynatrace/stackstate_checks/dynatrace/dynatrace.py

```diff
@@ -284,21 +384,22 @@ def _get_labels(self, dynatrace_component):
                 labels.append("managementZones:%s" % zone.get("name"))
         if dynatrace_component.entityId:
             labels.append(dynatrace_component.entityId)
-        if dynatrace_component.monitoringState:
+        if dynatrace_component.get('monitoringState'):
             if dynatrace_component.monitoringState.actualMonitoringState:
                 labels.append("actualMonitoringState:%s" % dynatrace_component.monitoringState.actualMonitoringState)
             if dynatrace_component.monitoringState.expectedMonitoringState:
                 labels.append(
                     "expectedMonitoringState:%s" % dynatrace_component.monitoringState.expectedMonitoringState)
-        for technologies in dynatrace_component.softwareTechnologies:
-            tech_label = ''
-            if technologies.get('type'):
-                tech_label += technologies['type']
-            if technologies.get('edition'):
-                tech_label += ":%s" % technologies['edition']
-            if technologies.get('version'):
-                tech_label += ":%s" % technologies['version']
-            labels.append(tech_label)
+        if dynatrace_component.get('softwareTechnologies'):
+            for technologies in dynatrace_component.softwareTechnologies:
+                tech_label = ''
```

Comment:
You can use something like: 
```
':'.join(filter(None, [technologies.get('type'), techno...]))
```
to clean this up a bit

Reply:
done.